recipes/llama_api_providers/examples_with_aws/Prompt_Engineering_with_Llama_3_On_Amazon_Bedrock.ipynb

```diff
@@ -0,0 +1,2358 @@
+      "Requirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.10/site-packages (from requests<3,>=2->langchain) (3.3.2)\n",
+      "Requirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.10/site-packages (from requests<3,>=2->langchain) (3.6)\n",
+      "Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/conda/lib/python3.10/site-packages (from requests<3,>=2->langchain) (1.26.18)\n",
+      "Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.10/site-packages (from requests<3,>=2->langchain) (2024.2.2)\n",
+      "Requirement already satisfied: greenlet!=0.4.17 in /opt/conda/lib/python3.10/site-packages (from SQLAlchemy<3,>=1.4->langchain) (3.0.3)\n",
+      "Requirement already satisfied: mypy-extensions>=0.3.0 in /opt/conda/lib/python3.10/site-packages (from typing-inspect<1,>=0.4.0->dataclasses-json<0.7,>=0.5.7->langchain) (1.0.0)\n"
+     ]
+    }
+   ],
+   "source": [
+    "# install packages\n",
+    "!python3 -m pip install -qU boto3\n",
+    "!python3 -m pip install langchain\n",
+    "\n",
+    "import boto3\n",
+    "import json "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from getpass import getpass\n",
+    "from urllib.request import urlopen\n",
+    "from typing import Dict, List\n",
+    "from langchain_community.llms import Bedrock\n",
+    "from langchain.chains import ConversationChain\n",
+    "from langchain.memory import ConversationBufferMemory\n",
+    "from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler\n",
+    "from langchain.memory import ChatMessageHistory\n",
+    "from langchain.schema.messages import get_buffer_string\n",
+    "import os\n",
+    "\n",
+    "# bedrock = boto3.client(service_name='bedrock', region_name='us-east-1')\n",
+    "# listModels = bedrock.list_foundation_models(byProvider='meta')\n",
+    "# print(\"\\n\".join(list(map(lambda x: f\"{x['modelName']} : { x['modelId'] }\", listModels['modelSummaries']))))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 315,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "LLAMA3_70B_INSTRUCT = \"meta.llama3-70b-instruct-v1:0\"\n",
+    "LLAMA3_8B_INSTRUCT = \"meta.llama3-8b-instruct-v1:0\"\n",
+    "# We'll default to the smaller 8B model for speed; change to LLAMA3_70B_CHAT for more advanced (but slower) generations\n",
+    "DEFAULT_MODEL = LLAMA3_8B_INSTRUCT\n",
+    "\n",
+    "\n",
+    "def completion(\n",
+    "    prompt: str,\n",
+    "    model: str = DEFAULT_MODEL,\n",
+    "    temperature: float = 0.6,\n",
+    "    top_p: float = 0.9,\n",
+    ") -> str:\n",
+    "    llm = Bedrock(model_id=model)\n",
+    "    prompt_str = ''.join(prompt)\n",
+    "    if not prompt_str.endswith(\"<|eot_id|><|start_header_id|>assistant<|end_header_id|>\"):\n",
```

Comment:
Lets extract this to a variable, and name appropriately so developers knows why we are doing this check explicitly? (We are doing this to determine if the agent must respond back or not. Is that correct?)